upgrade/helper/storage_version_migrator.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2023 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upgrade
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/pkg/apiextensions/storageversion"
+)
+
+// performs crd storage version upgrade
+// lists all the resources and,
+// keeps only one storage version on the crd
```

Comment:
I don't think, this is what we wanted to keep on CR i.e. only 1 version

What I know about the issue is - the fresh installation and old installation can have different version, but we need to update the CRD by removing the old versions and using the new versions. We can't delete and create again, as it will remove user CRs.

Reply:
@piyush-garg My understanding is,
* storage version should contain latest version and may contain old version too
* from the `migrate` package [1], first updating all the resources to latest version
* then keeps only the latest version under `status.storedVersions`

[1] - https://github.com/knative/pkg/blob/main/apiextensions/storageversion/migrator.go#L49-L56

Reply:
i am not sure if this will work like this, because just changing the version will not work for tekton resources, spec also needs to handled accordingly.

what i was expecting from this story was to handle the change of this field https://github.com/tektoncd/pipeline/blob/main/config/300-pipeline.yaml#L27